pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -73,9 +83,49 @@ type logInfo struct {
 	protocolStr string // protocol of the traffic logged
 }
 
+// logDedupRecord will be used as 1 sec buffer for log deduplication
+type logDedupRecord struct {
+	count       int64       // record count of duplicate log
+	initTime    time.Time   // initial time upon receiving packet log
+	bufferTimer *time.Timer // 1 sec buffer for each log
+}
+
+// logSafeGuard includes a map of log buffers and a r/w mutex for accessing the map
+type logRecordDedupMap struct {
+	logMutex sync.Mutex
+	logMap   map[string]*logDedupRecord
+}
+
+// deleteLogKey logs and deletes the log record in logDeduplication by logMsg.
+func (a *AntreaPolicyLogger) terminateLogKey(logMsg string) {
+	a.logDeduplication.logMutex.Lock()
+	defer a.logDeduplication.logMutex.Unlock()
+	<-a.logDeduplication.logMap[logMsg].bufferTimer.C
+	if a.logDeduplication.logMap[logMsg].count == 1 {
+		a.anpLogger.Printf(logMsg)
+	} else {
+		a.anpLogger.Printf(fmt.Sprintf("%s [%d packets in %s]", logMsg, a.logDeduplication.logMap[logMsg].count, time.Since(a.logDeduplication.logMap[logMsg].initTime)))
+	}
+	delete(a.logDeduplication.logMap, logMsg)
+}
+
+// updateLogKey initiates record or increases the count in logDeduplication corresponding to given logMsg.
+func (a *AntreaPolicyLogger) updateLogKey(logMsg string) bool {
+	a.logDeduplication.logMutex.Lock()
+	defer a.logDeduplication.logMutex.Unlock()
+	_, exists := a.logDeduplication.logMap[logMsg]
+	if exists {
+		a.logDeduplication.logMap[logMsg].count++
+	} else {
+		record := logDedupRecord{1, time.Now(), time.NewTimer(time.Second)}
```

Comment:
you want to make this delay configurable so that you can reduce it for unit tests